pkg/logger/logger_test.go

```diff
@@ -298,3 +298,114 @@ func TestErrorPath(t *testing.T) {
 		require.Nil(t, l)
 	})
 }
+
+func TestLoggerWith(t *testing.T) {
+	t.Parallel()
+
+	logStatements := func(l1 *SugarLogger) {
+		l1.Debug("debug message is logged")
+		l1.Info("info message is logged")
+		l1.Warn("warning message is logged")
+		l1.Error("error message is logged")
+	}
+
+	tests := []struct {
+		level               string
+		fileName            string
+		expectedMessages    []string
+		notExpectedMessages []string
+	}{
+		{
+			level:    "debug",
+			fileName: "/tmp/debug.txt",
```

Comment:
Can we use here some temp location and not go direct to /tmp folder

Reply:
Done